fleet_adapter_mir_actions/fleet_adapter_mir_actions/rmf_wait_until.py

```diff
@@ -0,0 +1,485 @@
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import importlib
+import requests
+from urllib.error import HTTPError
+
+from fleet_adapter_mir_actions.mir_action import MirAction, MirActionFactory
+from fleet_adapter_mir.robot_adapter_mir import ActionContext
+
+
+class ActionFactory(MirActionFactory):
+    def __init__(self, context: ActionContext):
+        MirActionFactory.__init__(self, context)
+        self.custom_modules = {}
+        # Raise error if config file is invalid
+        supported_signal_types = {'mission', 'plc', 'custom'}
+        signals = context.action_config.get('signals')
+        if signals is None:
+            # If the user did not provide any valid signal config, log a
+            # warning to remind users to provide signal type config in any task
+            # description they submit
+            self.context.node.get_logger().warn(
+                f'WaitUntil ActionFactory is instantiated for robot '
+                f'[{self.context.name}], but no valid signals config has been '
+                f'provided in the action config! Any move off signal config '
+                f'will have to be provided in the task description submitted '
+                f'to RMF.'
+            )
+            return
+
+        for signal_name, signal_config in signals.items():
+            signal_type = signal_config.get('signal_type')
+            if signal_type is None:
+                raise KeyError(
+                    f'WaitUntil MirAction signal config requires a '
+                    f'defined signal_type, but signal_type is not '
+                    f'provided for {signal_name}! Unable to instantiate '
+                    f'WaitUntil MirActionFactory.'
+                )
+            elif signal_type not in supported_signal_types:
+                raise ValueError(
+                    f'WaitUntil MirAction signal config requires a '
+                    f'defined signal_type, but signal_type provided '
+                    f'{signal_name} is not supported! We currently '
+                    f'support the following signal types: '
+                    f'{supported_signal_types}. Unable to instantiate '
+                    f'WaitUntil MirActionFactory.'
+                )
+
+            # Pass signal config into verify functions to validate types
+            # and catch missing values
+            match signal_type:
```

Comment:
I notice there are many places in this code where it does `match signal_type`.

Since we already support custom signal types via plugins, I would've expected that we could use the base class (I think `MoveOff`) to define all the interfaces needed across different signal types. With that we could have

```python
class WaitForMission(MoveOff):
    ...

class WaitForPLC(MoveOff):
    ...
```

and we would only need this one `match signal_type:` to instantiate `self.signals[signal_name] = ` for whichever signal type was asked for.